lfshttp/standalone/standalone.go

```diff
@@ -101,6 +101,11 @@ func gitDirAtPath(path string) (string, error) {
 	}
 	env = env[:n]
 
+	// Trim any trailing .git path segment.
+	if filepath.Base(path) == ".git" {
+		path = filepath.Dir(path)
+	}
+
```

Comment:
So I think this solves our present problem.  However, in the general case, I think our testsuite is telling us something: that we're passing a remote value that isn't valid.  If we have a bare repository, then that's the equivalent of the `.git` directory; there isn't a `.git` directory inside it, and so we're right to reject it.

I understand that it's nice to maintain backward compatibility here, but I don't think it makes sense for us to pass a URL that isn't valid over to the adapter and then fix it up.  I wonder if we could try a different approach: calling `os.Stat` on the path with the `.git` component in `EndpointFromLocalPath` and using it if it's valid, and omitting it if it's not.  That would preserve backwards compatibility while still making sure we have valid data.

Reply:
I'm happy to do that instead, or as well, since you advise it!  I figure this change, in itself, is effective and might as well stay, but having a belt-and-suspenders approach and fixing the output from `EndpointFromLocalPath()` is good with me.  I'll take a look at that next.

Reply:
OK, I've tried to implement your suggestion, @bk2204 -- see how that sits with you now.